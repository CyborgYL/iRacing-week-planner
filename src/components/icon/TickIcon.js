import React from 'react';
import styles from '../../styles/main.module.scss';

export default function TickIcon() {
  return <span className={`${styles.glyphicon} ${styles['glyphicon-ok']}`} />;
}
