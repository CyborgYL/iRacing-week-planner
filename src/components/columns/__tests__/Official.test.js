import React from 'react';
import { shallow } from 'enzyme';
import { describe, test } from '@jest/globals';
import TickIcon from '../../icon/TickIcon';
import { Official } from '../index';

describe('components/columns/Official', () => {
  test('renders correctly', () => {
    const component = shallow(<Official race={{ official: true }} />);

    expect(component.find(TickIcon).length).toBe(1);
    expect(component).toMatchSnapshot();
  });

  test('renders false correctly', () => {
    const component = shallow(<Official race={{ official: false }} />);

    expect(component.find(TickIcon).length).toBe(0);
    expect(component).toMatchSnapshot();
  });
});
