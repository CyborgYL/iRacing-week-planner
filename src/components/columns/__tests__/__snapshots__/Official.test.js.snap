// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/columns/Official renders correctly 1`] = `
<td>
  <div>
    <TickIcon />
  </div>
</td>
`;

exports[`components/columns/Official renders false correctly 1`] = `
<td>
  <div />
</td>
`;
